packages/monitoring-system/data-processor/src/data-processors/github-data-processor.ts

```diff
@@ -12,10 +12,175 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
-import {DataProcessor} from './data-processor-abstract';
+import {DataProcessor, ProcessorOptions} from './data-processor-abstract';
+import {Octokit} from '@octokit/rest';
+import {WriteResult} from '@google-cloud/firestore';
+import crypto from 'crypto';
+import {
+  GitHubRepositoryDocument,
+  OwnerType,
+  getRepositoryPrimaryKey,
+  GitHubEventDocument,
+  UNKNOWN_FIRESTORE_VALUE,
+} from '../firestore-schema';
+const {ORG, USER, UNKNOWN} = OwnerType;
 
+export interface GitHubProcessorOptions extends ProcessorOptions {
+  octokit?: Octokit;
+}
+
+interface GitHubEventResponse {
+  type: string;
+  repo: {
+    name: string;
+  };
+  actor: {
+    login: string;
+  };
+  payload: {};
+  created_at: string;
+  org?: {};
+  user?: {};
+}
+
+/**
+ * Collects and processes Events data from GitHub
+ */
 export class GitHubProcessor extends DataProcessor {
+  private octokit: Octokit;
+
+  constructor(options?: GitHubProcessorOptions) {
+    super(options);
+    this.octokit = options?.octokit || new Octokit();
+  }
+
+  /**
+   * Collect and process GitHub Events data
+   */
   public async collectAndProcess(): Promise<void> {
-    throw new Error('Method not implemented.');
+    return new Promise((resolve, reject) => {
+      this.listRepositories()
+        .then(repos => {
+          return Promise.all(
+            repos.map(repo => {
+              return this.listPublicEventsForRepository(repo);
```

Comment:
this method might require pagination, not sure if you're concerned with getting all the results!

Reply:
You're right! the first page fetches only the last 100 events, but given that this process will run at most every 5 mins and considering the current rate of events being generated, I think we should be fine with just the first page of results. But I'll add a note for the future when we're generating events at a higher rate :)